test/swarming/collect.py

```diff
@@ -132,10 +131,8 @@ def main():
         f.write(json.dumps(results, sort_keys=True, indent=2))
 
     #### Exit code
-    # Return non-zero if the task failed, but silence failures due to expiration
-    # of an internal failure, because we do not want Swarming infrastructure
-    # failures to be reported as build failures.
-    if status == 'pass' or status == 'expired' or status == 'internal_failure':
+    # Return non-zero if the task failed, but silence failures due to expiration:
+    if status == 'pass' or status == 'expired':
```

Comment:
What do we do now, when we have internal failure in task_result[]?

Reply:
Good point!

This field did not seem to be present in the JSON returned by the (updated) LUCI tools. I asked the LUCI team, see the discussion in b/159313821, but basically such fields are not present unless there is actually an internal failure (or a regular failure, etc).

I've updated this PR to properly check for field existence before their value, and the check for internal_failure is back.